server/src/test/java/org/elasticsearch/indices/DateMathIndexExpressionsIntegrationIT.java

```diff
@@ -41,15 +48,42 @@
 import static org.hamcrest.Matchers.notNullValue;
 
 public class DateMathIndexExpressionsIntegrationIT extends ESIntegTestCase {
+    private ZonedDateTime now;
+
+    @Before
+    public void setNow() {
+        now = ZonedDateTime.now(ZoneOffset.UTC);
+    }
+
+    /**
+     * the internal cluster calls System.nanoTime() and System.currentTimeMillis() during evaluations of requests
+     * that need date-math index resolution. These are not mockable in these tests. As is, executing requests as-is
+     * in these test cases can potentially result in invalid responses when day-boundaries are hit mid test run. Instead
+     * of failing when index resolution with `now` is one day off, this method wraps calls with the assumption that
+     * the day did not change during the test run.
+     */
+    public <Q extends ActionRequest, R extends ActionResponse> R dateSensitiveGet(ActionRequestBuilder<Q, R> builder) {
+        Runnable dayChangeAssumption = () -> assumeTrue("day changed between requests",
+            ZonedDateTime.now(ZoneOffset.UTC).getDayOfYear() == now.getDayOfYear());
+        R response;
+        try {
+            response = builder.get();
+        } catch (IndexNotFoundException e) {
+            // index resolver throws this if it does not find the exact index due to day changes
+            dayChangeAssumption.run();
```

Comment:
the reason this is here is because the System.nanotime's day that is picked up in a search request's start-time can differ from the indices that were created, which will cause indices to be resolved to ones that do not exist. This means that the dayChange occurred between the time of the first dayChangeAssumption.run(), and the time the search had begun. Although this time window is super small, could still happen and happened to me once during testing.

Reply:
when this occurs, I set a dayChangeAssumption check to run before throwing the exception so that if the error is because of the day change, I want the test to be ignored. If there was an IndexNotFoundException thrown and the day didn't change, then there is a real problem and the exception should be re-thrown.

does that make sense?

Reply:
Yes, thank you for making it clear. I think this is a good change and we should have this merged before we make an attempt to refactor this in a more general way